src/main/java/Storage.java

```diff
@@ -55,11 +55,14 @@ public void updateTasks(TaskList taskList) throws DukeException {
             FileWriter fileWriter = new FileWriter(file);
             for (Task task : taskList.getStoredTasks()) {
                 if (task instanceof ToDo) {
-                    fileWriter.write(String.join(SPLITTER, TaskType.TODO.getInitial(), getDoneString(task), task.getDescription()));
+                    fileWriter.write(String.join(SPLITTER, TaskType.TODO.getInitial(), getDoneString(task),
+                            task.getDescription()));
                 } else if (task instanceof Deadline) {
-                    fileWriter.write(String.join(SPLITTER, TaskType.DEADLINE.getInitial(), getDoneString(task), task.getDescription(), ((Deadline) task).getBy()));
+                    fileWriter.write(String.join(SPLITTER, TaskType.DEADLINE.getInitial(), getDoneString(task),
```

Comment:
Any reason why the string is joined this way?

Reply:
This is an interesting suggestion! I initially implemented it this way, but changed it as I felt that tasks should not have the responsibility of returning storage valid formats. Thank you so much for your helpful and insightful comments! 👍